subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/DaemonParameters.java

```diff
@@ -148,10 +148,14 @@ public Map<String, String> getSystemProperties() {
     }
 
     public Map<String, String> getEffectiveSystemProperties() {
+        return getEffectiveSystemProperties(System.getProperties());
+    }
+
+    public Map<String, String> getEffectiveSystemProperties(Map<?, ?> customSystemProperties) {
         Map<String, String> systemProperties = new HashMap<String, String>();
+        GUtil.addToMap(systemProperties, customSystemProperties);
```

Comment:
Is this the right order? it seems like this would allow immutable and mutable system properties to overwrite these custom ones?

Reply:
It's the other way around. If `customProperties` is added last, then it would allow clients to overwrite immutable system properties.